jedi/inference/star_args.py

```diff
@@ -22,7 +22,15 @@ def _iter_nodes_for_param(param_name):
     from jedi.inference.arguments import TreeArguments
 
     execution_context = param_name.parent_context
-    function_node = execution_context.tree_node
+    # Walk up the parso tree to get the FunctionNode we want. We use the parso
+    # tree rather than going via the execution context so that we're agnostic of
+    # the specific scope we're evaluating within (i.e: module or function,
+    # etc.).
+    # - .tree_name is a Name
+    # - .parent is a Param
+    # - .parent is a PythonNode(parameters)
+    # - .parent is the FunctionNode we want.
+    function_node = param_name.tree_name.parent.parent.parent
```

Comment:
Oh wow, that's the reason. IMO we should probably use `tree.search_ancestor(param_name.tree_name, 'funcdef', 'lambdef')`. That's probably more stable. If something ever changes (or maybe if lambdas are involved or something similar, some assumptions might break here.

Not sure about lambdas anyway.

Reply:
`tree.search_ancestor` definitely sounds more reliable! I'll look at using that.

Yeah, previously the module-scope things were getting `Module` as their `function_node`.

Reply:
ab2eb570a81922546f7c870f2d3c34878a55f642